### PR TITLE
Update to PyQt6

### PR DESCRIPTION
**Pull request by JoshuaAppleby**

### CODE REVIEW

**Comment by aawdls** [comment]
I cannot see any problems with this. Does it work?

**Comment by JoshuaAppleby** [comment]
Yes it does.

File: src/dls_response_matrix/respm_app.py [hunk 1]
```diff
@@ -3,7 +3,8 @@
 
 import cothread
 import pytac  # noqa
-from PyQt5 import QtCore, QtWidgets, uic  # noqa
+from PyQt6 import uic
```
Comment (MichaelStubbings) [inline src/dls_response_matrix/respm_app.py:6]: Does this get used at all?

Comment (JoshuaAppleby) [reply src/dls_response_matrix/respm_app.py:6]: yep, second line in the __init__ of MainWindow: uic.loadUi(UI_FILENAME, self)

File: pyproject.toml [hunk 2]
```diff
@@ -13,12 +13,13 @@ classifiers = [
 ]
 description = "Generator of storage ring response matrix"
 dependencies = [
-    "cothread>=2.18.2",
+    "cothread>=2.18.3",
```
Comment (MichaelStubbings) [inline pyproject.toml:16]: Is this and setuptools Qt dependent? It's not really worth doing anything if not, but sneaking version updates into other commits can make it hard to find if it causes any problems / needs to be reverted.

File: src/dls_response_matrix/respm_app.py (hunk 1 above)
Comment (MichaelStubbings) [reply src/dls_response_matrix/respm_app.py:6]: ah yeah, for some reason ctrl+f didnt pick it up

File: pyproject.toml (hunk 2 above)
Comment (JoshuaAppleby) [reply pyproject.toml:16]: Yes, so pyqt5 -> 6 requires a change to cothread. And the cothread updates require a minimum version of setuptools.

Comment (MichaelStubbings) [reply pyproject.toml:16]: perfect :)

**Review by aawdls — APPROVED** [review]

**Pull request merged**